### PR TITLE
ROX-21569: Add repo mapping URL to indexer config

### PR DESCRIPTION
**Pull request by daynewlee**
## Description
The indexer must get repository mapping data from correct URL:
indexer in security cluster should send request to Senor endpoint
indexer in central cluster should send request to Central endpoint

## Checklist
- [ ] Investigated and inspected CI test results

### CODE REVIEW

File: image/templates/helm/shared/config-templates/scanner-v4/indexer-config.yaml.tpl [hunk 1]
```diff
@@ -21,6 +21,8 @@ indexer:
       client_encoding=UTF8
     password_file: /run/secrets/stackrox.io/secrets/password
   get_layer_timeout: 1m
```
Comment (daynewlee) [inline image/templates/helm/shared/config-templates/scanner-v4/indexer-config.yaml.tpl:23]: @mclasmeier 
I am wondering if we could make this config to be conditional:
for indexer in secured cluster, we want to use sensor endpoint: `https://sensor.stackrox.svc/scanner/definitions`
for indexer in central cluster, we want to use central endpoint: `https://central.stackrox.svc/api/extensions/scannerdefinitions`

Comment (mclasmeier) [reply image/templates/helm/shared/config-templates/scanner-v4/indexer-config.yaml.tpl:23]: Hi! See https://github.com/stackrox/stackrox/pull/9225/files

File: image/templates/helm/shared/config-templates/scanner-v4/indexer-config.yaml.tpl [hunk 2]
```diff
@@ -21,6 +21,13 @@ indexer:
...
+  {{- if ._rox.env.centralServices }}
+  repository_to_cpe_url: https://central.stackrox.svc/api/extensions/scannerdefinitions?type=repo2cpe
```
Comment (RTann) [inline image/templates/helm/shared/config-templates/scanner-v4/indexer-config.yaml.tpl:25]: @mclasmeier is something like 

```
https://central.{{ .Release.Namespace }}.svc/api/extensions/scannerdefinitions?type=repo2cpe
```

possible? If so, let's switch all of these to equivalents

Comment (mclasmeier) [reply image/templates/helm/shared/config-templates/scanner-v4/indexer-config.yaml.tpl:25]: yes

Comment (daynewlee) [reply image/templates/helm/shared/config-templates/scanner-v4/indexer-config.yaml.tpl:25]: > @mclasmeier is something like
> 
> ```
> https://central.{{ .Release.Namespace }}.svc/api/extensions/scannerdefinitions?type=repo2cpe
> ```
> 
> possible? If so, let's switch all of these to equivalents

Is there a need for the sensor URL to do the same?

Comment (RTann) [reply image/templates/helm/shared/config-templates/scanner-v4/indexer-config.yaml.tpl]: @daynewlee yes, sensor will need the same. We can't assume the namespace will be `stackrox`

File: pkg/helm/charts/tests/centralservices/testdata/helmtest/scanner-v4.test.yaml [hunk 3]
```diff
@@ -224,3 +224,10 @@ tests:
     .serviceaccounts["scanner-v4"] | assertThat(. != null)
     .serviceaccounts["scanner-v4"] | .imagePullSecrets[] | select(.name == "stackrox")
     .serviceaccounts["scanner-v4"] | .imagePullSecrets[] | select(.name == "stackrox-scanner")
+
+- name: "indexer should be using central endpoints when deployed as part of central-services"
```
Comment (RTann) [inline pkg/helm/charts/tests/centralservices/testdata/helmtest/scanner-v4.test.yaml:228]: @mclasmeier is there a way to test with a non-stackrox namespace?

**Review by RTann — APPROVED** [review]
LGTM, but with one question

**Pull request merged**